app/tapes-data/[tape_id]/route.ts

```diff
@@ -41,41 +42,65 @@ const getTapePayload = async (tapeId:string):Promise<VerifyPayloadProxy> => {
   return replayLogs[0];
 }
 
+interface FullTapePayload {
+  tape?: string,
+  incard?: string,
+  args?: string,
+  entropy?: string
+}
 
 export async function GET(request: NextRequest, { params }: { params: { tape_id: string }}) {
     const tapeId = params.tape_id;
-    let data: Uint8Array = new Uint8Array();
+    // let data: Uint8Array = new Uint8Array();
+    let data: FullTapePayload = {};
     try {
       const tape = await getTapePayload(tapeId);
-      const rule = await getRule(tape.rule_id);
-      const abi = ethers.utils.defaultAbiCoder;
-      const entropy = `0x${generateEntropy(tape._msgSender,rule.id)}`;
+      const rule = await getRule(ruleIdFromBytes(tape.rule_id));
+      // const entropy = `0x${generateEntropy(tape._msgSender,rule.id)}`;
+      const entropy = generateEntropy(tape._msgSender,rule.id);
+
+      const inputData: FormatInCardPayload = {rule_id:rule.id};
+      if (tape.incard && tape.incard > 0) inputData.in_card = ethers.utils.hexlify(tape.incard);
+      if (tape.tapes && tape.tapes.length > 0) inputData.tapes = tape.tapes;
+      const incard = await formatInCard(inputData, {
+        cartesiNodeUrl: envClient.CARTESI_NODE_URL,
+        decode:true,
+        decodeModel:"bytes",
+        method:"POST"
+      });
 
-      data = ethers.utils.arrayify(abi.encode(
-        ["bytes","string","bytes","bytes32"],
-        [tape.tape,rule.args,rule.in_card,entropy]
-      ));
+      data = {
+        entropy: entropy,
+        tape: btoa(new Uint8Array(ethers.utils.arrayify(tape.tape)).reduce((data, byte) => data + String.fromCharCode(byte), '')),
+        incard: btoa(new Uint8Array(ethers.utils.arrayify(incard)).reduce((data, byte) => data + String.fromCharCode(byte), '')),
+        args: rule.args
+      };
```

Comment:
It seems like **incard** can be null, it happened in the test that I did here, so I am suggesting adding it to **data** after verifying it.
```suggestion
      data = {
        entropy: entropy,
        tape: btoa(new Uint8Array(ethers.utils.arrayify(tape.tape)).reduce((data, byte) => data + String.fromCharCode(byte), '')),
        //incard: btoa(new Uint8Array(ethers.utils.arrayify(incard)).reduce((data, byte) => data + String.fromCharCode(byte), '')),
        args: rule.args
      };
      if (incard) {
        data.incard = btoa(new Uint8Array(ethers.utils.arrayify(incard)).reduce((data, byte) => data + String.fromCharCode(byte), ''));
      }
```

Reply:
Ok, we can add another test
But the response from the inspect can be empty, but it won't be null.